### PR TITLE
Add docker image release workflow

### PR DESCRIPTION
**Pull request by milosgajdos**
The secrets used in this workflow have been manually added by me to GitHub secrets on this repo. They will be changed once we sort out Docker Hub accounts and tokens, but the current creds work fine for pushing images to Docker Hub.

The idea is when we tag a release the image is automatically built from the tag ref and pushed to Docker Hub.

Signed-off-by: Milos Gajdos <milosgajdos83@gmail.com>

### CODE REVIEW

**Review by chrispat — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: Build release image and push to hub
```
Comment (wy65701436) [inline .github/workflows/release.yaml:1]: the name will show in the action page, can we give an short one?

**Review by wy65701436 — APPROVED** [review]
lgtm

File: .github/workflows/release.yaml (hunk 1 above)
Comment (milosgajdos) [reply .github/workflows/release.yaml:1]: Yes, we can! What would you suggest? `Release Image`?

Comment (wy65701436) [reply .github/workflows/release.yaml]: `Release Image`  is OK, you can merge it after revise.

**Pull request merged**